### ISSUE TITLE
Add support for Log4j2

### ISSUE THREAD

**Issue by sdeleuze** [oracle/graalvm-reachability-metadata#50]
I currently see a ` java.lang.NoSuchMethodException: org.apache.logging.log4j.message.DefaultFlowMessageFactory.<init>()` error in Spring smoke tests [logging-log4j2](https://github.com/spring-projects/spring-aot-smoke-tests/tree/main/logging-log4j2) sample.

**Comment by sdeleuze** [comment]
I tried to provide support for this but have found a blocking issue, so I think native support for Log4j2 needs to happen directly on library side rather than here, see [related comment](https://issues.apache.org/jira/browse/LOG4J2-2649?focusedCommentId=17610488&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17610488) on their bugtracker.

**Comment by sdeleuze** [comment]
Given the fact that Log4j2 needs to happen at least partially via builtin support, I think that's better to close this issue at least for now. Upcoming discussion should happen on https://issues.apache.org/jira/browse/LOG4J2-2649.

**Comment by darxriggs** [comment]
It might be useful to update the link to the "logging-log4j2" sample in the issue description to a version of the repository that still contains this sample (now that it has been removed).

**Comment by sdeleuze** [comment]
I have linked the sample attached to the Log4j2 issue.

**Comment by Periecle** [comment]
@sdeleuze I believe I have found a way to do that without changing anything into Log4j2. They need LambdaMetafactory thing to work around the Security Context switching issue. However, as Graal Native Image does not have a security manager - we do not need it, so I just replaced that class via SVM Target/Substitute functionality. I verified my idea and provided a simple PoC for that. I have generated a basic reflect-config and manually edited it to support as many features as I could, but that is not a production-ready solution, and still, it works.

Here is my PoC. I am not sure we can merge such a thing in that repo(because of NativeServiceLoaderUtil).
https://github.com/Periecle/log4j2-native-tests
